src/lib/Syntax.hs

```diff
@@ -1015,22 +1134,32 @@ data EnvQuery =
  | SourceNameInfo   SourceName
    deriving (Show, Generic)
 
+data SourceNameDef n =
+  -- The Nothing case of the UVar means the definition had an error in it.
+  -- The Nothing case of the ModuleSourceName means that the variable was
+  -- defined in the current module.
+  SourceNameDef (Maybe (UVar n)) (Maybe ModuleSourceName)
```

Comment:
Yes, they're slightly different. For one thing, the source map that the module itself uses might contain more names, since we might only export a subset of the names. (Currently we do export all the names because I haven't added syntax for declaring an export list, but I'll do that t some point.) But also they need to be different because the error messages can be different if a source name is ambiguous etc.